### ISSUE TITLE
sh doesn't work with Go programs

### ISSUE THREAD

**Issue by juanfra684** [amoffat/sh#361]
Sorry if the bug is known but I didn't find any reference in the bug tracker or google.

```
$ cat test.py 
import sh

ctest = sh.Command("/tmp/test/ctest")
gotest = sh.Command("/tmp/test/gotest")

print("C:", ctest())
print("Go:", gotest())
$ cat ctest.c 
#include<stdio.h>

int main() {
    printf("OK\n");
    return 0;
}
$ cat gotest.go 
package main

func main() {
	print("OK\n")
}
$ ./ctest 
OK
$ ./gotest 
OK
$ python3.6 test.py
C: OK

Go:
$ cat test-fg.py 
import sh

ctest = sh.Command("/tmp/test/ctest")
gotest = sh.Command("/tmp/test/gotest")

print("C:", ctest(_fg=True))
print("Go:", gotest(_fg=True))
$ python3.6 test-fg.py
OK
C: None
OK
Go: None
```

**Comment by amoffat** [comment]
https://golang.org/pkg/builtin/#print

> The print built-in function formats its arguments in an implementation-specific way and **writes the result to standard error.** Print is useful for bootstrapping and debugging; it is not guaranteed to stay in the language.

```python
import sh
gotest = sh.Command("/tmp/test/gotest")
print("Go:", gotest(_err_to_out=True))
```

**Comment by juanfra684** [comment]
You're right, with `_err_to_out=True` works fine. However, that was just an example and I though that the problem was the same that I was seeing in the real app.

```
$ export GOPATH=`pwd`
$ go get -u github.com/grantseltzer/golumns
$ cat gotest.go 
package main

import "github.com/grantseltzer/golumns"

func main() {
	golumns.Display([]string{"OK", "OK"})
}
$ go build gotest.go
$ python3.6 test.py                  
C: OK

Go:
```

I don't know if this is the correct solution but this code works:

```
$ cat test-io.py 
import sh
from io import StringIO

buf = StringIO()

ctest = sh.Command("/tmp/test/ctest")
gotest = sh.Command("/tmp/test/gotest")

ctest(_out=buf)
gotest(_out=buf)

print("C:", buf.getvalue())
print("Go:", buf.getvalue())
$ python3.6 test-io.py      
C: OK

Go: OK

```

Thanks for the help.

**Comment by amoffat** [comment]
@juanfra684 I just tried the following go program and it worked fine, because `fmt.Printf` goes to stdout

```go
package main
import "fmt"

func main() {
    fmt.Printf("OK\n")
}
```

However, I noticed something in the golumns library:  https://github.com/GrantSeltzer/golumns/blob/master/display.go#L13-L16  The condition can cause an early return if there is no TTY attached to stdin.  In sh, by default, [no TTY](http://amoffat.github.io/sh/sections/special_arguments.html#tty-in) is attached to stdin.

So try using `_tty_in=True` on the command that was failing you.  I bet the golumns library won't fail then.
